src/URI.js

```diff
@@ -1240,15 +1240,19 @@
       this._parts = URI.parse(String(href), this._parts);
     } else if (_URI || _object) {
       var src = _URI ? href._parts : href;
+      var hasQueryPart = false;
       for (key in src) {
         if (key === 'query') {
+          hasQueryPart = true;
```

Comment:
wait a second, this can be true even if `src['query']` is undefined, isnt it?

Reply:
you can keep the `if (key === 'query') { continue; }` part and add `if (src.query) { this.query(src.query, false); }`. you only need to execute .query() if there actually is a value (other than null, undefined, empty-string)